drake/solvers/MultipleTreeProblem.m

```diff
@@ -0,0 +1,463 @@
+classdef MultipleTreeProblem
+  
+  properties
+    robot
+    endEffectorId
```

Comment:
even robot/endEffector could be generalized (easily?) to systems.  The endpoint, etc, could be taken in as a constraint handle?